app/models/repository_file.rb

```diff
@@ -56,7 +59,29 @@ def initialize(*args, &block)
 
   def save!
     raise RecordNotSavedError unless valid?
-    repository.save_file(temp_file.path, target_path, message, user)
+    repository.save_file(source_file.path, target_path, message, user)
+  end
+
+  def file_upload_type
+    @file_upload_type || 'local'
+  end
+
+  def source_file
+    case file_upload_type
+    when 'local'
+      temp_file
+    when 'remote'
+      @remote_file ||= retrieve(remote_file_iri)
+    end
+  end
+
+  def source_filename
+    case file_upload_type
+    when 'local'
+      temp_file.original_filename
+    when 'remote'
+      remote_file_iri.split('/').last
```

Comment:
- ends on `/`: we don't care as `split` will take care of this (a lone `/` at the end will just be dropped)
- contains query-string: i will fix that.